pkg/broker/variables.go

```diff
@@ -87,3 +92,43 @@ func (bv *BrokerVariable) ToSchema() map[string]interface{} {
 
 	return schema
 }
+
+// ValidateVariables validates a list of BrokerVariables adhere to their JSONSchema.
+func ValidateVariables(parameters map[string]interface{}, schemaVariables []BrokerVariable) error {
+
+	allErrors := &multierror.Error{
+		ErrorFormat:utils.LineErrorFormatter,
+	}
+
+	for _, variable := range schemaVariables {
+
+		value, ok := parameters[variable.FieldName]
+		if !ok {
+			// According to json schema, the required property trumps the default value.
+			if variable.Required {
+				multierror.Append(allErrors, fmt.Errorf("missing required parameter \"%s\"", variable.FieldName))
+				continue
+			} else if variable.Default != nil {
+				// Insert the default value into the parameters
+				value = variable.Default
```

Comment:
I don't think the defaults will always end up being valid because some might be templates that are executed.

Reply:
Currently defaults are applied to missing variables, then the whole map is ran through the json schema validation.